### PR TITLE
feat: enable emoji buttons in chat input

### LINKED ISSUE #225: [Chat] User cannot click on emoji
OS:  Android
Device: Mobile
TMail version:  2.1.3

### *Reproduce Step:
1, User open a conversation
2, User tap on Emoji icon

### **Actual:** 
Nothing happen when user click on Emoji icon

https://user-images.githubusercontent.com/124866146/248578120-83cbb11a-f74e-4bed-833c-7bceebf5cdbc.mp4

### **Expected:** 
- Open system emoji

### PR DESCRIPTION
**Pull request by imGok**
demoji
This Closes #225

### CODE REVIEW

File: lib/widgets/twake_components/twake_icon_button.dart [hunk 1]
```diff
@@ -50,13 +50,16 @@ class TwakeIconButton extends StatelessWidget {
...
+            child: icon != null
+                ? Icon(
+                    icon,
+                    size: size,
+                    fill: fill,
+                    weight: weight,
+                  )
+                : imagePath != null
+                    ? SvgPicture.asset(imagePath!)
+                    : null,
```
Comment (hoangdat) [inline lib/widgets/twake_components/twake_icon_button.dart:62]: Indent = 2 spaces

**Comment by hoangdat** [comment]
What happens for the case: `pick emoji, then type some text`. Please demo

**Comment by imGok** [comment]
> What happens for the case: `pick emoji, then type some text`. Please demo

added in the description!

**Review by sherlockvn — APPROVED** [review]

File: lib/pages/chat/chat_input_row.dart [hunk 2]
```diff
@@ -170,9 +170,9 @@ class ChatInputRow extends StatelessWidget {
                                 );
                               },
                               child: TwakeIconButton(
-                                paddingAll: controller.inputText.isEmpty ? 5.0: 12,
+                                paddingAll: controller.inputText.isEmpty ? 5.0 : 12,
                                 tooltip: "Emojis",
```
Comment (nqhhdev) [inline lib/pages/chat/chat_input_row.dart:174]: pls move it to `L10n`

File: lib/pages/chat/chat_view.dart [hunk 3]
```diff
@@ -269,58 +269,70 @@ class ChatView extends StatelessWidget {
...
-                                          ),
-                                          TextButton.icon(
-                                            style: TextButton.styleFrom(
-                                              padding: const EdgeInsets.all(16),
+                                    ? Padding(
+                                        padding: EdgeInsets.only(
+                                          bottom: bottomSheetPadding,
+                                          left: bottomSheetPadding,
+                                          right: bottomSheetPadding,
+                                        ),
```
Comment (nqhhdev) [inline lib/pages/chat/chat_view.dart:283]:
```suggestion
                                    ? Padding(padding: bottomSheetPadding)
```

**Comment by hoangdat** [comment]
> added in the description!

not friendly enough for the user, a lot of touch and switch screen. Please use the system emoji of keyboard, dont use emoji picker from Fluffy. Thanks

**Comment by hoangdat** [comment]
- Plan A: try to find the way to turn on the search for current emoji 
- Plan B: find the full package for emoji picker

File: lib/pages/chat/chat_view.dart (hunk 3 above)
Comment (imGok) [reply lib/pages/chat/chat_view.dart:283]: bottomSheetPadding is a double value, I can't use it there and I don't want a top padding aswell

**Comment by imGok** [comment]
Plan A : Possible, will need to do a whole new widget for this, implementing our own emoji database (bad idea), use EmojiPickerUtils.search 
Plan B : I found this : https://pub.dev/packages/flutter_emoji_gif_picker 

We should keep an eye on this one, it's for native keyboard and only support iOS for now : https://pub.dev/packages/keyboard_emoji_picker

**Comment by imGok** [comment]
Add https://pub.dev/packages/keyboard_emoji_picker for iOS only : 

- library is not compatible with our version of flutter_webrtc 

image
flutter_webrtc must be downgraded in order to install the library

Solutions : 
- downgrade flutter_webrtc
image
https://stackoverflow.com/a/61394530

- wait for an update from emoji keyboard library

**Comment by nqhhdev** [comment]
WDYT about the lib: https://pub.dev/packages/emoji_picker_flutte.

I recommend for youuuu

**Comment by hoangdat** [comment]
it is not a critical issue in the app. I propose the workaround solution first.
- use the current emoji picker (fluffy) for android, ios, web.
We will come back later in this topic

**Comment by hoangdat** [comment]
Additional: 
- How to switch from Emoji Picker to Keyboard more smooth, easier to type text?

**Comment by hoangdat** [comment]
please resolve all comment and rebase

**Review by hoangdat — APPROVED** [review]

**Pull request merged**